pkg/azure/minter.go

```diff
@@ -236,11 +236,34 @@ func (credMinter *AzureCredentialsMinter) AssignResourceScopedRole(ctx context.C
 		return fmt.Errorf("more than one role %q found", targetRole)
 	}
 
+	credMinter.logger.Debugf("getting current role assignments for service principal %s", principalName)
+	filter := fmt.Sprintf("principalId eq '%s'", principalID)
+	currentRoleAssignments, err := credMinter.roleAssignmentsClient.List(ctx, filter)
+	if err != nil {
+		credMinter.logger.WithError(err).Error("failed to list role assignments")
+		return err
+	}
+
 	for _, resourceGroup := range resourceGroups {
 		scope := "subscriptions/" + credMinter.subscriptionID + "/resourceGroups/" + resourceGroup
+
+		// check whether assignment already exists
+		alreadyExists := false
+		for _, r := range currentRoleAssignments {
+			if strings.Contains(*r.Properties.Scope, scope) {
```

Comment:
Can you add a comment here clarifying why we can't do an eq and what the data strings look like? I see scope above but what does r.Properties.Scope come out looking like? The test data looks awfully similar.

Reply:
the Contains() comparison is because our generated scope var string doesn't have a leading /, and the returned r.Properties.Scope does have the leading /.

i reworked the generated string a few lines above to just put the leading / in the string so we can now do a simple string comparison. putting the leading / still allows all the minting to work.